migration/src/androidTest/kotlin/kin/sdk/migration/MigrationManagerIntegrationTest.kt

```diff
@@ -113,220 +130,290 @@ class MigrationManagerIntegrationTest {
+        // create account which we will backup
+        val accountToBackup: KinAccountCoreImpl = createActivateAndFundOldKinAccount() as KinAccountCoreImpl
+        val accountToBackupBalance = accountToBackup.balanceSync
+        val passphrase = "qwerty123"
+        val export = accountToBackup.export(passphrase)
+        // create regular old account which will now be the active account.
+        val oldAccountToMigrate = createActivateAndFundOldKinAccount()
+        // migrate this account
+        val kinClientAfterMigration = getKinClientOnNewKinBlockchain()
+        // verify that the migrated account is indeed the current active account
+        assertThat(oldAccountToMigrate?.publicAddress, `is`(equalTo(kinClientAfterMigration?.getAccount(kinClientAfterMigration.accountCount - 1)?.publicAddress)))
+        val importAccount = kinClientAfterMigration?.importAccount(export, passphrase)
+        // verify that the after the restore then the backup account is now indeed the current active account
+        assertThat(accountToBackup.publicAddress, `is`(equalTo(kinClientAfterMigration?.getAccount(kinClientAfterMigration.accountCount - 1)?.publicAddress)))
+        assertFailsWith(AccountNotFoundException::class) {
+            importAccount?.balanceSync
+        }
+        // migrate backup account
+        getKinClientOnNewKinBlockchain()
+        val balanceSync = importAccount?.balanceSync
+        // verify that after migration the balance of the backup account is equal to the balance he had before restore and migrate
+        assertEquals(balanceSync?.value()?.compareTo(accountToBackupBalance.value()), 0)
+    }
 
-        })
-        assertTrue(latch.await(timeoutDurationSecondsLong, TimeUnit.SECONDS))
+    @Test
+    @LargeTest
+    fun start_AlreadyMigrated_ClearMigratedFlag_ZeroBalance_AccountNotActivated_getNewKinClient() {
+        start_AlreadyMigrated_ClearMigratedFlag_getNewKinClient(false)
     }
 
+    //TODO differentiate between those 2 tests using the events, meaning add verify to events.
+
     @Test
     @LargeTest
-    fun start_AlreadyMigratedButNotSavedLocally_WithBalance_getNewKinClient() {
+    fun start_AlreadyMigratedBut_ClearMigratedFlag_WithBalance_getNewKinClient() {
+        start_AlreadyMigrated_ClearMigratedFlag_getNewKinClient(true)
+    }
+
+    private fun start_AlreadyMigrated_ClearMigratedFlag_getNewKinClient(withBalance: Boolean) {
+        val migrationDidStart = AtomicBoolean()
+        val isNewSdk = AtomicBoolean()
+        var error: Error? = null
         val latch = CountDownLatch(1)
-        getNewKinClientAfterMigration()// blocking call
-        teardown()
-        fail("remove this fail after finished with the TODO")//TODO  add funding with faucet
+        val kinClient = getNewKinClientAfterMigration()
+        removeData()
+        if (withBalance) { //TODO do i really want to fund the new migrated account???
```

Comment:
don't really get why the balance is matters at all? and if so test only with or only without.
moreover, additional important tests would be to verify you got back the new balance on kin3

Reply:
Yep, I need to change this test, thats why I added the TODO.
Will keep it there because going to do it in a dedicated branch.